### DISCUSSION TITLE
focusOptions object documentation

### DISCUSSION THREAD

**Discussion by shestakov-vladyslav** [react-hook-form/react-hook-form#9167] [Ideas]
Is it poosible to add more details on focusOptions object in documentation, please?
![image](https://user-images.githubusercontent.com/18069756/194613043-deccf146-48c8-4af2-9616-cf54f2c2fde1.png)

**Comment by bluebill1049** [comment]
yes, we can. Are you using typescreipt? it should include in the type definition with autocomplete.

**Comment by juan-velvet** [comment]
I just found myself trying to see what is FocusOptions  but there is no documentation at all, last resource was to check the codebase.

**Comment by trandaison** [comment]
@shestakov-vladyslav Here is the type
```ts
export type FieldArrayMethodProps = {
    shouldFocus?: boolean;
    focusIndex?: number;
    focusName?: string;
};
```
I am using typescript, but somehow the type autocomplete does not work.

**Comment by Arjun-Mavai** [comment]
I propose adding the `**shouldFocus**` option to the `append` method of `useFieldArray` in React Hook Form. This addition will greatly benefit developers by providing enhanced control over UI behavior, especially in complex forms. When set to `false`, it prevents the newly appended field from automatically gaining focus, ensuring a smoother user experience and avoiding disruptive UI jumps. This feature, although not widely documented, can significantly improve form interactions and usability in React applications. Adding it to the official documentation would make this valuable option more accessible to the developer community.

const { fields, append } = useFieldArray({ name: "yourFieldArray", control });

// Add a new field without auto-focusing

const addField = ( ) => {
  append({ description:0 } , {  shouldFocus: false });
};

@bluebill1049

**Reply by patelmilanun** [reply]
can u add it in the DOC if u got time and raise a PR, what do u say @bluebill1049

**Comment by gajraj-gan** [comment]
This is still not added, what is the difference between focusIndex and focusName?
